### PR TITLE
Tests related to brute force attack prevention.

### PR DESCRIPTION
**Pull request by lukasz-soszynski-eliatra**
Signed-off-by: Lukasz Soszynski <lukasz.soszynski@eliatra.com>

### Description
[Describe what this change achieves]
* Category (Integration tests)
* Why these changes are required?
* What is the old behaviour before changes and new behaviour after changes?

Test related to brute-force attack prevention and a minor correction for TlsTests.

### Issues Resolved
[List any issues this PR will resolve]

Is this a backport? If so, please add backport PR # and/or commits #

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by DarshitChanpura** [comment]
Bwc tests are broken on main. Tracking issue here: https://github.com/opensearch-project/security/issues/2221

File: src/integrationTest/java/org/opensearch/security/IpBruteForceAttacksPreventionTests.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	}
+
+	@Test
+	public void shouldBlockIpAddress() {
+		authenticateUserWithIncorrectPassword(CLIENT_IP_3, USER_2, ALLOWED_TRIES);
+		try(TestRestClient client = cluster.createGenericClientRestClient(userWithSourceIp(USER_2, CLIENT_IP_3))) {
+
+			HttpResponse response = client.getAuthInfo();
+
+			response.assertStatusCode(SC_UNAUTHORIZED);
```
Comment (DarshitChanpura) [inline src/integrationTest/java/org/opensearch/security/IpBruteForceAttacksPreventionTests.java:79]: Out of curiosity, how do you test that an IP is blocked after multiple failed logins

Comment (DarshitChanpura) [inline src/integrationTest/java/org/opensearch/security/IpBruteForceAttacksPreventionTests.java:79]: Out of curiosity, how do you test that an IP is blocked after multiple failed logins? I mean, that end user will still see UNAUTHORIZED error but on server side how do we know that the IP is blocked?

File: src/integrationTest/java/org/opensearch/test/framework/cluster/CloseableHttpClientFactory.java [hunk 2]
```diff
@@ -50,6 +57,9 @@ public CloseableHttpClient getHTTPClient() {
 			.setDefaultSocketConfig(SocketConfig.custom().setSoTimeout(60, TimeUnit.SECONDS).build())
 			.build();
 		hcb.setConnectionManager(cm);
+		if(routePlanner != null) {
+			hcb.setRoutePlanner(routePlanner);
+		}
```
Comment (DarshitChanpura) [inline src/integrationTest/java/org/opensearch/test/framework/cluster/CloseableHttpClientFactory.java:62]: out of curiosity, why do we need routePlanner?

Comment (lukasz-soszynski-eliatra) [reply src/integrationTest/java/org/opensearch/test/framework/cluster/CloseableHttpClientFactory.java:62]: In order to set the request source IP address, this is done by `org.opensearch.test.framework.cluster.LocalAddressRoutePlanner` which implements `HttpRoutePlanner`. 

The Apache HTTP Client in version 4 contained a convenient method to set the request source IP address `RequestConfig.custom().setLocalAddress(inetAddress).build();`. But the method was removed in version 5 of Apache HTTP Client

File: src/integrationTest/java/org/opensearch/security/IpBruteForceAttacksPreventionTests.java (hunk 1 above)
Comment (lukasz-soszynski-eliatra) [reply src/integrationTest/java/org/opensearch/security/IpBruteForceAttacksPreventionTests.java:84]: I added an explicit check of logs in the commit https://github.com/opensearch-project/security/pull/2245/commits/c95fbdbcb3b37788551cd1eaa015d44904954c79

Comment (DarshitChanpura) [reply src/integrationTest/java/org/opensearch/security/IpBruteForceAttacksPreventionTests.java:84]: nice

**Review by DarshitChanpura — APPROVED** [review]
LGTM!

File: src/integrationTest/java/org/opensearch/security/IpBruteForceAttacksPreventionTests.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+		try(TestRestClient client = cluster.createGenericClientRestClient(userWithSourceIp(USER_1, CLIENT_IP_7))) {
+
+			HttpResponse response = client.getAuthInfo();
+
+			response.assertStatusCode(SC_OK);
+		}
+	}
+
+	@Test
+	public void shouldBlockIpWhenFailureAuthenticationCountIsGraterThanAllowedTries() {
```
Comment (cwperks) [inline src/integrationTest/java/org/opensearch/security/IpBruteForceAttacksPreventionTests.java:123]: nit: _Greater_

File: src/integrationTest/java/org/opensearch/security/UserBruteForceAttacksPreventionTests.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+				HttpResponse response = client.getAuthInfo();
+
+				response.assertStatusCode(SC_UNAUTHORIZED);
+		}
+		//Rejecting REST request because of blocked user:
+		logsRule.assertThatContain("Rejecting REST request because of blocked user: " + USER_2.getName());
+	}
+
+	@Test
+	public void shouldBlockUserWhenNumberOfFailureLoginAttemptIsGraterThanLimit() {
```
Comment (cwperks) [inline src/integrationTest/java/org/opensearch/security/UserBruteForceAttacksPreventionTests.java:82]: Same here.

**Review by cwperks — APPROVED** [review]
Looks good to me!

**Pull request merged**